aspnet/fundamentals/routing.rst

```diff
@@ -186,100 +370,56 @@ Adding a colon ``:`` after the name allows additional inline constraints to be s
     - Steve
     - Used to enforce that a non-parameter value is present during during URL generation.
 
-Inline constraints must match one of the above options, or an exception will be thrown.
-
-.. tip:: To constrain a parameter to a known set of possible values, you can use a regex: ``{action:regex(list|get|create)}``. This would only match the ``action`` route value to ``list``, ``get``, or ``create``. If passed into the constraints dictionary, the string "list|get|create" would be equivalent. Constraints that are passed in the constraints dictionary (not inline within a template) that don't match one of the known constraints are also treated as regular expressions.
+.. warning:: Route constraints that verify the URL can be converted to a CLR type (such as ``int`` or ``DateTime``) always use the invariant culture - they assume the URL is non-localizable.
```

Comment:
The most important detail here is this:

> The framework-provided route constraints do not modify the values stored in route values. All route values parsed from the URL will be stored as strings.

In your example, [this code] (https://github.com/aspnet/Routing/blob/dev/src/Microsoft.AspNetCore.Routing/Constraints/FloatRouteConstraint.cs#L53) will parse the string to make sure it's a valid `float`, but the resulting `float` value is thrown away.

Users often expect that they can do this, which won't work:
```
var price = (float)routeData.Values["price"];
```

We should also take care to make sure we're not using words like `price` with floating point types :trollface: